### PR TITLE
Fix new SSO connection regressions

### LINKED ISSUE #18: New AWS SSO config format
**Describe the bug**
This plugin is not compatible with the [new default AWS SSO sessions format](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-sso.html#sso-configure-profile-token)

**To Reproduce**
1. Define a new sso session profile with `aws configure sso`
2. Try to deploy a serverless stack with this plugin and the new profile
3. The plugin will try to use the default profile.

**Expected behavior**
The plugin loads the right profile.

**Desktop (please complete the following information):**
 - OS: MacOS
 - Version v1.1.2
 - Serverless Version 3.26.0

### PR DESCRIPTION
**Pull request by guiyom-e**
Fix new SSO connection regressions introduced in https://github.com/thomasmichaelwallace/serverless-better-credentials/pull/26
- refactor: simplify forEach loop in addSsoDataToProfiles
- fix: handle profiles from credentials file
- fix: add sso region when using new sso method
- refactor: simplify sso login code

Should close https://github.com/thomasmichaelwallace/serverless-better-credentials/issues/18, https://github.com/thomasmichaelwallace/serverless-better-credentials/issues/27 and https://github.com/thomasmichaelwallace/serverless-better-credentials/issues/28

### CODE REVIEW

**Comment by guiyom-e** [comment]
@tonyvdv and @alaexeyshustov this PR should fix your issues, may you check it ?

@thomasmichaelwallace v1.2.1 may be released I suggest!

File: src/SsoCredentials/__tests__/getSsoConfig.test.ts [hunk 1]
```diff
@@ -44,7 +45,6 @@ test('it loads the sso config from a file with the new SSO login', () => {
     `
       [testing]
       sso_start_url = https://testing.awsapps.com/start
```
Comment (MichaelLebrand) [inline src/SsoCredentials/__tests__/getSsoConfig.test.ts]: guess you can drop the `sso_start_url` as well, as it's defined by the `test-session` already

**Review by MichaelLebrand — APPROVED** [review]
small comment regarding test case, though current setup works.
Not the biggest TS guy so perhaps needs a second pair of eyes.

File: src/SsoCredentials/__tests__/getSsoConfig.test.ts (hunk 1 above)
Comment (guiyom-e) [reply src/SsoCredentials/__tests__/getSsoConfig.test.ts]: Thanks, it's done!

**Comment by MichaelLebrand** [comment]
@thomasmichaelwallace sorry to push you a bit on this but could we go ahead and merge this? Thanks!!

**Comment by tobim** [comment]
Is there any way to use this branch in the meantime?

**Comment by guiyom-e** [comment]
> Is there any way to use this branch in the meantime?

@tobim : you may clone the repo, install and build it (`yarn && yarn build`), and then link it to your own `package.json` instead of using the default npm version.

It can be done with a `yarn link` or a direct file link:
`"serverless-better-credentials": "file:[path to serverless-better-credentials folder]"`

**Comment by tobim** [comment]
@guiyom-e: Thanks, after following your steps and running `npm install` in my project `serverless` complains:
```
Serverless plugin "serverless-better-credentials" not found. Make sure it's installed and listed in the "plugins" section of your serverless config file. Run "serverless plugin install -n serverless-better-credentials" to install it.
```

**Comment by thomasmichaelwallace** [comment]
OK- although last time I did this it broke everything 😉

**Comment by thomasmichaelwallace** [comment]
Now released as `1.2.1`.

**Comment by MichaelLebrand** [comment]
> OK- although last time I did this it broke everything 😉

hehe fair point, will see if we run into anything in the coming days/weeks.
Thanks for pushing the buttons anyway!

**Pull request merged**